velox/functions/sparksql/tests/SortArrayTest.cpp

```diff
@@ -153,7 +153,8 @@ TEST_F(SortArrayTest, array) {
       makeNullableNestedArrayVector(reverseNested(expected)));
 }
 
-TEST_F(SortArrayTest, map) {
+// Map is unordered, so sorting is not supported.
+TEST_F(SortArrayTest, DISABLED_map) {
```

Comment:
Does Spark throw for map type? If so, perhaps we could verify the exception in unit test.

Reply:
Yeap, spark also only allows orderable types to sort, while map type is not orderable inside spark.

```scala
checkArgumentDataTypes() match {
      case TypeCheckResult.TypeCheckSuccess =>
        argument.dataType match {
          case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
//...
```

```sql
spark-sql> select sort_array(array(map(1,2)));
Error in query: cannot resolve 'sort_array(array(map(1, 2)), true)' due to data type mismatch: sort_array does not support sorting array of type map<int,int> which is not orderable; line 1 pos 7;
spark-sql> select array_sort(array(map(1,2)));
Error in query: cannot resolve '(namedlambdavariable() < namedlambdavariable())' due to data type mismatch: LessThan does not support ordering on type map<int,int>; line 1 pos 7;
```

Added the exception check in the unit tests